llvm/include/llvm/IR/IntrinsicInst.h

```diff
@@ -1249,6 +1252,21 @@ class MemSetInlineInst : public MemSetInst {
   }
 };
 
+/// This class wraps the llvm.memset.pattern intrinsic.
+class MemSetPatternInst : public MemSetInst {
+public:
+  ConstantInt *getLength() const {
+    return cast<ConstantInt>(MemSetInst::getLength());
```

Comment:
IIRC the constant length restriction has been removed.

See also https://discourse.llvm.org/t/rfc-introducing-an-llvm-memset-pattern-inline-intrinsic/79496/5.

Reply:
Yes this was a holdover from before I'd removed that restriction. Now removed.